Modules/_sqlite/cursor.c

```diff
@@ -416,15 +415,15 @@ begin_transaction(pysqlite_Connection *self)
     rc = sqlite3_finalize(statement);
     Py_END_ALLOW_THREADS
 
-    if (rc != SQLITE_OK && !PyErr_Occurred()) {
-        _pysqlite_seterror(self->db);
+    if (rc != SQLITE_OK) {
+        goto error;
     }
 
-error:
-    if (PyErr_Occurred()) {
-        return -1;
-    }
     return 0;
+
+error:
+    _pysqlite_seterror(self->db);
+    return -1;
```

Comment:
@serhiy-storchaka, @berkerpeksag: I'm a little bit unsure about this change. What if `Py_BEGIN_ALLOW_THREADS` or `Py_END_ALLOW_THREADS` fail?

Reply:
Never mind. I've reverted 393368f89cd6f4a02693d5e9e863bbbf9eb19498 for now, so the error handling is unchanged.